benchmarks/streamable.py

```diff
@@ -0,0 +1,232 @@
+            Mode.to_bytes: ModeParameter(to_bytes),
+            Mode.from_bytes: ModeParameter(FullBlock.from_bytes, to_bytes),
+            Mode.to_json: ModeParameter(FullBlock.to_json_dict),
+            Mode.from_json: ModeParameter(FullBlock.from_json_dict, FullBlock.to_json_dict),
+        },
+    ),
+}
+
+
+def run_for_ms(cb: Callable[[], Any], ms_to_run: int = 100) -> int:
+    iterations: int = 0
+    start = monotonic()
+    while int((monotonic() - start) * 1000) < ms_to_run:
+        cb()
+        iterations += 1
+    return iterations
+
+
+def calc_stdev(iterations: List[int]) -> float:
+    return 0 if len(iterations) < 2 else int(stdev(iterations) * 100) / 100
+
+
+def run_benchmarks(data: Data, mode: Mode, runs: int, milliseconds: int) -> None:
+    results: Dict[Data, Dict[Mode, List[int]]] = {}
+    for current_data, parameter in benchmark_parameter.items():
+        results[current_data] = {}
+        if data == Data.all or current_data == data:
+            print(f"\nRun {mode.name} benchmarks with the class: {parameter.data_class.__name__}")
+            print_row("runs", "ms/run", "mode", "avg iterations", "std deviation")
+            for current_mode, current_mode_parameter in parameter.mode_parameter.items():
+                results[current_data][current_mode] = []
+                if mode == Mode.all or current_mode == mode:
+                    iterations: int
+                    all_iterations: List[int] = results[current_data][current_mode]
+                    for _ in range(max(1, runs)):
+                        obj = parameter.object_creation_cb()
+                        if current_mode == Mode.creation:
+                            cls = type(obj)
+                            iterations = run_for_ms(lambda: cls(**obj.__dict__), milliseconds)
+                        else:
+                            assert current_mode_parameter is not None
+                            conversion_cb = current_mode_parameter.conversion_cb
+                            assert conversion_cb is not None
+                            if current_mode_parameter.preparation_cb is not None:
+                                obj = current_mode_parameter.preparation_cb(obj)
+                            iterations = run_for_ms(lambda: conversion_cb(obj), milliseconds)
+                        all_iterations.append(iterations)
+                        print_row("last", milliseconds, current_mode.name, iterations, calc_stdev(all_iterations), "\r")
+                    average_iterations: int = int(sum(all_iterations) / runs)
+                    print_row(runs, milliseconds, current_mode.name, average_iterations, calc_stdev(all_iterations))
+
+
+data_option_help: str = "|".join([d.name for d in Data])
+mode_option_help: str = "|".join([m.name for m in Mode])
+
+
+@click.command()
+@click.option("-d", "--data", default=Data.all.name, help=data_option_help)
+@click.option("-m", "--mode", default=Mode.all.name, help=mode_option_help)
```

Comment:
Do you want [`type=click.Choice`](https://click.palletsprojects.com/en/7.x/options/#choice-options)?

Reply:
Would have been nice to get support for enums added directly to click, but they declined.  https://github.com/pallets/click/issues/591#issuecomment-673179242  Though, there are a few patterns provided at https://github.com/pallets/click/issues/605.

Reply:
Thanks, they were all not super satisfying but i picked one of the solutions there to have it at least with `Enum` :)